src/main/scala/ch/epfl/bluebrain/nexus/admin/index/DistributedDataIndex.scala

```diff
@@ -0,0 +1,188 @@
+package ch.epfl.bluebrain.nexus.admin.index
+
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.cluster.Cluster
+import akka.cluster.ddata.LWWRegister.Clock
+import akka.cluster.ddata.Replicator._
+import akka.cluster.ddata.{LWWMapKey, _}
+import akka.pattern.ask
+import akka.util.Timeout
+import cats.effect.{Async, IO}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import ch.epfl.bluebrain.nexus.admin.exceptions.UnexpectedState
+import ch.epfl.bluebrain.nexus.admin.organizations.Organization
+import ch.epfl.bluebrain.nexus.admin.projects.Project
+import ch.epfl.bluebrain.nexus.admin.types.ResourceF
+import ch.epfl.bluebrain.nexus.commons.types.RetriableErr
+import ch.epfl.bluebrain.nexus.commons.types.search.Pagination
+
+import scala.collection.SortedSet
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Index implementation based on Akka Distributed Data.
+  *
+  * @param askTimeout         actor ask timeout
+  * @param consistencyTimeout distributed data the consistency timeout
+  */
+class DistributedDataIndex[F[_]](askTimeout: Timeout, consistencyTimeout: FiniteDuration)(implicit as: ActorSystem,
+                                                                                          F: Async[F])
+    extends Index[F] {
+
+  private implicit val node: Cluster = Cluster(as)
+
+  private implicit def orgClock[A]: Clock[ResourceF[A]] = { (_: Long, value: ResourceF[A]) =>
+    value.rev
+  }
+  private implicit val timeout: Timeout = askTimeout
+
+  private val replicator = DistributedData(as).replicator
+
+  private val labelToProjectsKey      = LWWMapKey[String, ResourceF[Project]]("labels_to_projects")
+  private val labelToOrganizationsKey = LWWMapKey[String, ResourceF[Organization]]("labels_to_orgs")
+  private val organizationsListKey    = LWWRegisterKey[RevisionedValue[SortedSet[ResourceF[Organization]]]]("orgs_list")
```

Comment:
I don't think you need the `LWWRegister`s at all... since you can get the whole maps and use the values... either way, when you read from the map you get the whole map (please correct me if I'm wrong).

This makes the code rely on only two maps, which if we think about it, they share the same kind of operations (put and getAll); everything else can be reimplemented on top of these. Here's an example of what I've done in IAM:
https://github.com/bogdanromanx/nexus-iam/blob/realms/src/main/scala/ch/epfl/bluebrain/nexus/iam/index/KeyValueStore.scala
and the tests:
https://github.com/bogdanromanx/nexus-iam/blob/realms/src/test/scala/ch/epfl/bluebrain/nexus/iam/index/KeyValueStoreSpec.scala

You don't need to refactor everything now, but at least drop the registers if they're not needed.

Reply:
if we just get the values from the map, we have to sort it every time we fetch it in order to list the organizations/projects in alphabetical order

Reply:
I think that's fine, better than using an additional two registers. Consider that the number of projects if fairly low...

Reply:
@bogdanromanx Do you think it's also better to use the map for the listing of projects by organization?

Reply:
Sure; something like: `projectsMap.values().filter(_.org == expectedOrg).sortBy(...)`